modal/stub.py

```diff
@@ -347,22 +345,11 @@ async def serve(self, client=None, stdout=None, show_progress=None, timeout=None
         """
         from ._watcher import AppChange, watch
 
-        if not is_local():
+        if self._app is not None:
             raise InvalidError(
-                "Can not run an app from within a container. You might need to do something like this: \n"
-                'if __name__ == "__main__":\n'
-                "    stub.serve()\n"
-            )
-
-        if self._app_id is not None:
-            raise InvalidError(
-                f"Found existing app '{self._app_id}'. You may have nested stub.serve() inside a running app like this:\n"
-                'if __name__ == "__main__":\n'
-                "    with stub.run():\n"
-                "        stub.serve() # ❌\n\n"
-                "You might need to do something like this: \n"
-                'if __name__ == "__main__":\n'
-                "    stub.serve()\n"
+                "The stub already has an app running."
```

Comment:
I took the liberty of removing the previous exceptions. I think they were great in the past before we had the `modal serve` command. But it's probably better to promote `modal serve` today.